python/semantic_kernel/kernel.py

```diff
@@ -641,6 +735,66 @@ def create_function_from_prompt(
 
         return function
 
+    def create_function_from_prompt_yaml(self, text: str, **kwargs: Any) -> "KernelFunction":
+        """
+        Create a KernelFunction object from a YAML prompt.
+
+        Args:
+            text (str): The YAML prompt text.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            KernelFunction: The created KernelFunction object.
+        """
+
+        import yaml
+
+        try:
+            yaml_data: Dict[str, Any] = yaml.safe_load(text)
+
+            prompt_template_config = PromptTemplateConfig(
+                name=yaml_data["name"],
+                description=yaml_data["description"],
+                template=yaml_data["template"],
+                template_format=yaml_data.get("template_format", "semantic-kernel"),
+                input_variables=yaml_data.get("input_variables", []),
+                execution_settings=yaml_data.get("execution_settings", {}),
+            )
+
+            prompt_execution_settings: Dict[str, PromptExecutionSettings] = kwargs.get(
+                "execution_settings", prompt_template_config.execution_settings
+            )
+
+            # TODO @jmj - The user that serializes the KernelFunction into YAML might not know the service_id ahead of time.
```

Comment:
I'm not sure if there is any kind of auto-resolution that should be done for YAML deserialized KernelFunctions. I would think that most of the time users would have the service as "default", but there is no guarantee that if someone else is reusing the deserialized YAML that they have a "default" service registered to their Kernel.

Should there be an attempt at deserialization or when the KernelFunction is invoked to try and autoresolve an AI service that doesn't match the name in the YAML file or leave it up to the user to reconfigure the YAML?